app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -260,46 +264,56 @@ class MEJSPlayer {
 
     this.node.innerHTML = markup;
 
-    // Handle 'canplay' events fired by player
+    // Bind to canplay event to switch off loading when media is ready
     this.mediaElement.addEventListener(
       'canplay',
       this.handleCanPlay.bind(this)
     );
 
-    this.player.buildquality(this.player, null, null, this.mediaElement);
-    this.player.buildtracks(this.player, null, this.player.layers, this.mediaElement);
-
     // Build playlists button from the new stream when not in playlists
     if(!playlistItemsT) {
       this.player.options.playlistItemDefaultTitle = this.currentStreamInfo.embed_title;
       this.player.buildaddToPlaylist(this.player, null, null, null);
     }
 
-    // Turn on captions
-    this.toggleCaptions();
+    // Initialize playlist items for the new stream
+    if(playlistItemsT) {
+      this.player.buildplaylistItems(this.player, null, null, this.mediaElement);
+    }
+
+    // Build quality
+    this.player.buildquality(this.player, null, null, this.mediaElement);
+
+    this.reInitializeCaptions(playlistItemsT);
 
     this.player.load();
-    if(isEnded) {
-      this.player.play();
-    }
+    this.player.play();
+  }
 
-    // Update markers in playlists
-    if(playlistItemsT) {
-      let promises = [];
-      const playlistIds = this.playlistItem
-      ? [this.playlistItem.playlist_id, this.playlistItem.id]
-      : [];
-      promises.push(this.mejsMarkersHelper.getMarkers(...playlistIds));
-      Promise.all(promises)
-        .then(() => {
-          this.mejsMarkersHelper.updateVisualMarkers();
-        })
-        .catch(error => {
-          console.log('Promise rejection error');
-        });
-      this.player.buildmarkers(this.player, this.player.controls, null, this.mediaElement);
+  /** Based on the availability of captions in currentStreamInfo build tracks
+   * in playlists and item page
+   * @function reInitializeCaptions
+   * @param playlistItemsT 
+   */
+  reInitializeCaptions(playlistItemsT) {
+    if (this.currentStreamInfo.captions_path) {
+      // Re-arrange player features when tracks are available
+      // in both playlists and item page
+      this.player.featurePosition.tracks = 6;
+      this.player.featurePosition.fullscreen = 9;
```

Comment:
This looks brittle.  I wonder if there is another way we could do this since I'm afraid this will break if we add other controls.  What do you think?

Reply:
This wouldn't break the code, but if we add new controls they would be out of order. And that is an issue.
This is the way I could think of to do this. But there should be a better way.